<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>587</width>
    <height>442</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QPushButton" name="buttonForward">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>90</y>
      <width>61</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>前进</string>
    </property>
   </widget>
   <widget class="QPushButton" name="buttonBack">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>200</y>
      <width>61</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>后退</string>
    </property>
   </widget>
   <widget class="QPushButton" name="buttonRight">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>140</y>
      <width>61</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>右转</string>
    </property>
   </widget>
   <widget class="QPushButton" name="buttonLeft">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>140</y>
      <width>61</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>左转</string>
    </property>
   </widget>
   <widget class="QWidget" name="">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>80</y>
      <width>131</width>
      <height>131</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <item>
      <widget class="QSlider" name="sliderKP">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSlider" name="sliderKI">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSlider" name="sliderKD">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>90</y>
      <width>41</width>
      <height>111</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QLabel" name="label">
       <property name="text">
        <string>KP:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>KI:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>KD:</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>80</y>
      <width>41</width>
      <height>131</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <item>
      <widget class="QLineEdit" name="lineKP">
       <property name="text">
        <string>0.0</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="lineKI">
       <property name="text">
        <string>0.0</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="lineKD">
       <property name="text">
        <string>0.0</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>587</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>buttonForward</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>GoForward()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>120</x>
     <y>157</y>
    </hint>
    <hint type="destinationlabel">
     <x>137</x>
     <y>82</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBack</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>GoBack()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>118</x>
     <y>261</y>
    </hint>
    <hint type="destinationlabel">
     <x>141</x>
     <y>351</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonRight</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>TurnRight()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>190</x>
     <y>210</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>297</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonLeft</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>TurnLeft()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>48</x>
     <y>208</y>
    </hint>
    <hint type="destinationlabel">
     <x>49</x>
     <y>309</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sliderKP</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>TransmitParam(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>349</x>
     <y>147</y>
    </hint>
    <hint type="destinationlabel">
     <x>410</x>
     <y>61</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineKP</sender>
   <signal>editingFinished()</signal>
   <receiver>MainWindow</receiver>
   <slot>TransmitParam()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>454</x>
     <y>136</y>
    </hint>
    <hint type="destinationlabel">
     <x>476</x>
     <y>96</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sliderKI</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>TransmitParam(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>387</x>
     <y>189</y>
    </hint>
    <hint type="destinationlabel">
     <x>455</x>
     <y>311</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineKI</sender>
   <signal>editingFinished()</signal>
   <receiver>MainWindow</receiver>
   <slot>TransmitParam()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>434</x>
     <y>187</y>
    </hint>
    <hint type="destinationlabel">
     <x>516</x>
     <y>196</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sliderKD</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MainWindow</receiver>
   <slot>TransmitParam(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>362</x>
     <y>228</y>
    </hint>
    <hint type="destinationlabel">
     <x>410</x>
     <y>349</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineKD</sender>
   <signal>editingFinished()</signal>
   <receiver>MainWindow</receiver>
   <slot>TransmitParam()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>433</x>
     <y>222</y>
    </hint>
    <hint type="destinationlabel">
     <x>507</x>
     <y>260</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>GoForward()</slot>
  <slot>GoBack()</slot>
  <slot>TurnRight()</slot>
  <slot>TurnLeft()</slot>
  <slot>TransmitParam(int)</slot>
  <slot>TransmitParam(QString)</slot>
  <slot>ChangeKpLine(int)</slot>
  <slot>TransmitParam()</slot>
 </slots>
</ui>
